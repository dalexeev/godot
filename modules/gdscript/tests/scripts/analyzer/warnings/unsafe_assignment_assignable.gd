var variant_int_var: Variant = 1


func my_func(_hard_int_param: int = variant_int_var) -> void:
	pass


func test():
	var hard_int: int = 1
	var variant_int: Variant = 1
	var weak_int = 1

	var _hard_int_1: int = hard_int
	var _hard_int_2: int = variant_int
	var _hard_int_3: int = weak_int

	var _variant_int_1: Variant = hard_int
	var _variant_int_2: Variant = variant_int
	var _variant_int_3: Variant = weak_int

	var _weak_int_1 = hard_int
	var _weak_int_2 = variant_int
	var _weak_int_3 = weak_int
