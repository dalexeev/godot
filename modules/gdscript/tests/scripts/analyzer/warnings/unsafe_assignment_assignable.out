GDTEST_OK
>> WARNING
>> Line: 4
>> UNSAFE_ASSIGNMENT
>> The parameter "_hard_int_param" is of the subtype "int" but a value of the supertype "Variant" was assigned.
>> WARNING
>> Line: 14
>> UNSAFE_ASSIGNMENT
>> The variable "_hard_int_2" is of the subtype "int" but a value of the supertype "Variant" was assigned.
>> WARNING
>> Line: 15
>> UNSAFE_ASSIGNMENT
>> The variable "_hard_int_3" is of the subtype "int" but a value of the supertype "Variant" was assigned.
